test/process_test.cpp

```diff
@@ -0,0 +1,262 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <algorithm>
+#include <array>
+#include <climits>
+#include <iostream>
+#include <random>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include <thread>
+
+#include <migraphx/env.hpp>
+#include <migraphx/file_buffer.hpp>
+#include <migraphx/process.hpp>
+#include <migraphx/filesystem.hpp>
+
+#ifndef _WIN32
+#include <cstring>
+#endif
+#include "test.hpp"
+
+#ifdef _WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <io.h>
+#include <fcntl.h>
+#undef getpid
+#define getpid _getpid
+#endif
+
+static migraphx::fs::path executable;
+static std::string string_data = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, "
+                                 "sed do eiusmod tempor incididunt ut labore et dolore magna "
+                                 "aliqua. Ut enim ad minim veniam, quis nostrud exercitation "
+                                 "ullamco laboris nisi ut aliquip ex ea commodo consequat. "
+                                 "Duis aute irure dolor in reprehenderit in voluptate velit "
+                                 "esse cillum dolore eu fugiat nulla pariatur. Excepteur sint "
+                                 "occaecat cupidatat non proident, sunt in culpa qui officia "
+                                 "deserunt mollit anim id est laborum.";
+
+std::vector<char> read_stdin()
+{
+    std::vector<char> result;
+    std::array<char, 1024> buffer{};
+    std::size_t len = 0;
+#ifdef _WIN32
+    if(_setmode(_fileno(stdin), _O_BINARY) == -1)
+        throw std::runtime_error{"failure setting IO mode to binary"};
+#endif
+    while((len = std::fread(buffer.data(), 1, buffer.size(), stdin)) > 0)
+    {
+        if(std::ferror(stdin) != 0 and std::feof(stdin) == 0)
+            throw std::runtime_error{std::strerror(errno)};
+
+        result.insert(result.end(), buffer.begin(), buffer.begin() + len);
+    }
+    return result;
+}
+
+std::string get_temp_file(std::string_view test_id)
+{
+    auto pid = getpid();
+    auto tid = std::this_thread::get_id();
+    auto clk = std::chrono::steady_clock::now().time_since_epoch().count();
+    std::stringstream ss;
+    ss << std::hex << "migx-test-" << pid << '-' << tid << '-' << clk << '-' << test_id << ".txt";
+    return ss.str();
+}
+
+TEST_CASE(string_stdin)
+{
+    auto out = migraphx::fs::temp_directory_path() / get_temp_file("string");
```

Comment:
Use `tmp_dir` to create the temp file. It will handle the cleanup automatically.